core/string/shared/succ.rb

```diff
@@ -50,6 +50,7 @@
     "9Z99z99Z".send(@method).should == "10A00a00A"
 
     "ZZZ9999".send(@method).should == "AAAA0000"
+    "/[]9999".send(@method).should == "/[]10000"
```

Comment:
I do like new specs!
I see you even have https://github.com/opal/opal/blob/master/spec/opal/core/date_spec.rb for instance, which would be very interesting to merge if it is compatible with MRI.